Sources/ClientRuntime/Telemetry/Metrics/MeterProvider.swift

```diff
@@ -0,0 +1,16 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// A Meter Provider provides implementations of Meters.
+public protocol MeterProvider {
+    /// Provides a Meter.
+    ///
+    /// - Parameter scope: the name of the instrumentation scope that uniquely identifies this meter
+    /// - Parameter attributes: instrumentation scope attributes to associate with emitted telemetry data
+    /// - Returns: a Meter
+    func getMeter(scope: String, attributes: Attributes?) -> Meter
```

Comment:
Q: This question applies to other things too like `TracerProvider`, but what's the reason for having this extra layer `XYZProvider` providing concrete implementation of `XYZ` that then returns concrete thing we want (e.g., instrument or span)? 

What're the benefits of doing this instead of having respective `XYZProvider` protocols contain the function signatures that return concrete instance of what `XYZ` returns (in this case, the 6 or so "entry points" for creating different types of instruments)?

Reply:
The `Provider` pattern gives telemetry authors the chance to handle implementation details, such as concurrency and instance management.

For example, one customer may want different implementations of `Meter` per client, but another may just want to share a singleton across clients.

> (in this case, the 6 or so "entry points" for creating different types of instruments)?

For `Meter` specifically, this standardizes the types of instruments per the [OpenTelemetry Metrics API](https://opentelemetry.io/docs/specs/otel/metrics/api/#meter-operations) to encourage reusable implementations.